main/src/ca/uwaterloo/flix/language/errors/WeederError.scala

```diff
@@ -699,28 +699,27 @@ object WeederError {
   }
 
   /**
-    * An error raised to indicate that a newly defined type (or alias) has a
-    * pre-defined name.
+    * An error raised to indicate that a newly defined name reserved.
     *
-    * @param ident the name that conflicts.
+    * @param ident the reserved name that conflicts.
     * @param loc   the location where the error occurred.
     */
-  case class ReservedTypeName(ident: Name.Ident, loc: SourceLocation) extends WeederError {
-    def summary: String = "Re-definition of a pre-defined type name."
+  case class ReservedName(ident: Name.Ident, loc: SourceLocation) extends WeederError {
```

Comment:
Is it clear to programmers what a reserved name is? 

Should we call it ReservedKeyword instead? or similar?

Reply:
Keyword doesn't quite work because it could be a keyword or operator. I guess we could have one error for each?